lib/cisco_node_utils/cmd_ref/fabricpath.yaml

```diff
@@ -0,0 +1,178 @@
+# fabricpath
+---
+# Fabricpath feature is not available on N3K and N9K
+_exclude: [/N3/, /N9K/]
+
+aggregate_multicast_routes:
+  _exclude:
+    - /N5K/
+  kind: boolean
+  config_get: "show run fabricpath"
+  config_get_token: '/^fabricpath multicast aggregate\-routes$/'
+  config_set: "%s fabricpath multicast aggregate-routes"
```

Comment:
My code already had a whole bunch of printf-style configs. Wherever there are platform differences, I have tried to use key-value style

From: Mike Wiebe <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-node-utils <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Friday, November 13, 2015 at 7:57 AM
To: cisco/cisco-network-node-utils <cisco-network-node-utils@noreply.github.com<mailto:cisco-network-node-utils@noreply.github.com>>
Cc: Cisco Employee <dcherian@cisco.com<mailto:dcherian@cisco.com>>
Subject: Re: [cisco-network-node-utils] adding fabricpath topo and fixed feature_set as a class method (#75)


In lib/cisco_node_utils/cmd_ref/fabricpath.yaml<https://github.com/cisco/cisco-network-node-utils/pull/75#discussion_r44797760>:

> @@ -0,0 +1,178 @@
> +# fabricpath
> +---
> +# Fabricpath feature is not available on N3K and N9K
> +_exclude: [/N3/, /N9K/]
> +
> +aggregate_multicast_routes:
> +  _exclude:
> +    - /N5K/
> +  kind: boolean
> +  config_get: "show run fabricpath"
> +  config_get_token: '/^fabricpath multicast aggregate\-routes$/'
> +  config_set: "%s fabricpath multicast aggregate-routes"


What is the rational for using printf-style vs. key-value wildcard substitution? Using Key-value wildcards are more flexible in terms of multiple platform support.

https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/cmd_ref/README_YAML.md#wildcard-substitution

-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-node-utils/pull/75/files#r44797760>.